scripts/token-list.js

```diff
@@ -0,0 +1,59 @@
+// Creates a new JSON file that is compatible with Uniswap's token list spec.
+// See https://uniswap.org/blog/token-lists/ for details.
+//
+// This script is called just before publishing so the created file is updated
+// and included in the published package.
+
+'use strict'
+
+const fs = require('fs')
+
+const packageJson = require('../package.json')
+const metadata = require('../src/vesper-metadata.json')
+
+const tokenList = {
+  name: 'Vesper Tokens',
+  timestamp: new Date().toISOString(),
+  version: {
+    major: Number.parseInt(packageJson.version.split('.')[0]),
+    minor: Number.parseInt(packageJson.version.split('.')[1]),
+    patch: Number.parseInt(packageJson.version.split('.')[2])
+  },
+  tokens: []
+    .concat(metadata.pools, metadata.tokens)
+    .map(function (item) {
+      const {
+        address,
+        chainId,
+        decimals,
+        logoURI,
+        name,
+        poolName,
+        stage,
+        symbol,
+        type
+      } = item
+      const tags = []
+      if (type) {
+        tags.push('pool', type)
+      }
+      if (stage === 'retired') {
+        tags.push('retired')
+      }
+      return {
+        address,
+        chainId,
+        decimals,
+        logoURI,
+        name: poolName || name,
+        symbol,
+        tags: tags.length ? tags : undefined
```

Comment:
if no tags are set, what do you think about setting the empty array `[]` ? That way, whoever uses this code can just do `tags.map()`, `tags.slice()` or similar functions, instead of having to validate that the array is defined with values

Reply:
Good point. But since `tags` is optional, any user code will have to check for that prop anyway.